pennylane/transforms/decompositions/clifford_t/solovay_kitaev.py

```diff
@@ -0,0 +1,312 @@
+        return qml.math.eye(2, dtype=complex), qml.math.eye(2, dtype=complex)
+
+    # The angle phi comes from the Eq. 10 in the Solovay-Kitaev algorithm paper (arXiv:0505030)
+    phi = 2.0 * qml.math.arcsin(qml.math.sqrt(qml.math.sqrt((0.5 - 0.5 * qml.math.cos(theta / 2)))))
+
+    # Begin decomposition by computing the rotation operations V and W
+    v = qml.RX(phi, [0])
+    w = qml.RY(2 * math.pi - phi, [0]) if axis[2] > 0 else qml.RY(phi, [0])
+
+    # Get the similarity transormation matrices S and S.dag
+    ud = qml.math.linalg.eig(matrix)[1]
+    vwd = qml.math.linalg.eig(qml.matrix(v @ w @ v.adjoint() @ w.adjoint()))[1]
+    s = ud @ qml.math.conj(qml.math.transpose(vwd))
+    sdg = vwd @ qml.math.conj(qml.math.transpose(ud))
+
+    # Get the required matrices V' and W'
+    v_hat = s @ v.matrix() @ sdg
+    w_hat = s @ w.matrix() @ sdg
+
+    return w_hat, v_hat
+
+
+def sk_decomposition(op, depth, basis_set=(), basis_length=10):
+    r"""Approximate an arbitrary single-qubit gate in the Clifford+T basis using the `Solovay-Kitaev algorithm <https://arxiv.org/abs/quant-ph/0505030>`_.
+
+    This method implements a recursive Solovay-Kitaev decomposition that approximates any :math:`U \in \text{SU}(2)`
+    operation with :math:`\epsilon > 0` error up to a global phase. Increasing the recursion ``depth`` should
+    reduce this error. In general, this algorithm runs in :math:`O(\text{log}^{2.71}(1/\epsilon))` time and produces
+    a decomposition with :math:`O(\text{log}^{3.97}(1/\epsilon))` operations.
+
+    Args:
+        op (~pennylane.operation.Operation): A single-qubit gate operation
+        depth (int): Depth until which the recursion occurs
+        basis_set (list(str)): Basis set to be used for the decomposition and building an approximate set internally.
+            It accepts the following gate terms: ``['X', 'Y', 'Z', 'H', 'T', 'T*', 'S', 'S*']``, where `*` refers
+            to the gate adjoint. Default value is ``['T', 'T*', 'H']``
+        basis_length (int): Maximum expansion length of Clifford+T sequences in the internally built approximate set.
+            Default is `10`
+
+    Returns:
+        list(~pennylane.operation.Operation): A list of gates in the Clifford+T basis set that approximates the given operation
+
+    Raises:
+        ValueError: If the given operator acts on more than one wires
+
+    **Example**
+
+    Suppose one would like to decompose :class:`~.RZ` with rotation angle :math:`\phi = \pi/3`:
+
+    .. code-block:: python3
+
+        import numpy as np
+        import pennylane as qml
+
+        op  = qml.RZ(np.pi/3, wires=0)
+
+        # Get the gate decomposition in ['T', 'T*', 'H']
+        ops = qml.transforms.decompositions.sk_decomposition(op, depth=5)
+
+        # Get SU2 matrix from the ops
+        op_matrix = qml.prod(*reversed(ops)).matrix()
+        su2_matrix = op_matrix / np.sqrt((1 + 0j) * np.linalg.det(op_matrix))
```

Comment:
:thinking: I'm confused about the phase

Reply:
We now separately return the phase information.